public/js/pimcore/object/tags/localizedfields.js

```diff
@@ -289,6 +289,12 @@ pimcore.object.tags.localizedfields = Class.create(pimcore.object.tags.abstract,
                                 globalLanguageIndex = this.frontendLanguages.indexOf(language);
                             }
                             this.countrySelect.setValue(this.frontendLanguages[globalLanguageIndex]);
+
+                            this.availablePanels[this.currentLanguage].hide();
+                            this.component.updateLayout();
+                            this.availablePanels[language].show();
+                            this.currentLanguage = language;
+                            this.component.updateLayout();
```

Comment:
maybe extract the logic to a method and call it in `select` and `pimcoreGlobalLanguageChanged` event listeners.